code/modules/mob/living/simple_animal/hostile/jungle_mobs.dm

```diff
@@ -82,4 +82,241 @@
+	var/turf/T = get_turf(src)
+	..()
+	new /obj/structure/leaper_bubble(T)
+
+/obj/effect/temp_visual/leaper_projectile_impact
+	name = "leaper bubble"
+	icon = 'icons/obj/projectiles.dmi'
+	icon_state = "leaper_bubble_pop"
+	layer = ABOVE_ALL_MOB_LAYER
+	duration = 3
+
+/obj/effect/temp_visual/leaper_projectile_impact/Initialize()
+	. = ..()
+	new /obj/effect/decal/cleanable/leaper_sludge(get_turf(src))
+
+/obj/effect/decal/cleanable/leaper_sludge
+	name = "leaper sludge"
+	desc = "A small pool of sludge, containing trace amounts of leaper venom"
+	icon = 'icons/effects/tomatodecal.dmi'
+	icon_state = "tomato_floor1"
+
+/obj/structure/leaper_bubble
+	name = "leaper bubble"
+	desc = "A floating bubble containing leaper venom, the contents are under a surprising amount of pressure."
+	icon = 'icons/obj/projectiles.dmi'
+	icon_state = "leaper"
+	obj_integrity = 10
+	max_integrity = 10
+	density = FALSE
+
+/obj/structure/leaper_bubble/Initialize()
+	. = ..()
+	float(on = TRUE)
+	QDEL_IN(src, 100)
+
+/obj/structure/leaper_bubble/Destroy()
+	new /obj/effect/temp_visual/leaper_projectile_impact(get_turf(src))
+	playsound(src,'sound/effects/snap.ogg',50, 1, -1)
+	..()
+
+/obj/structure/leaper_bubble/Crossed(atom/movable/AM)
+	if(isliving(AM))
+		var/mob/living/L = AM
+		if(!istype(L, /mob/living/simple_animal/hostile/jungle/leaper))
+			playsound(src,'sound/effects/snap.ogg',50, 1, -1)
+			L.Knockdown(50)
+			if(iscarbon(L))
+				var/mob/living/carbon/C = L
+				C.reagents.add_reagent("leaper_venom", 5)
+			qdel(src)
+	..()
+
+/datum/reagent/toxin/leaper_venom
+	name = "Leaper venom"
+	id = "leaper_venom"
+	description = "A toxin spat out by leapers that while harmless in small doses, quickly creates a toxic reaction if too much is in the body."
+	color = "#801E28" // rgb: 128, 30, 40
+	toxpwr = 0
+	taste_description = "frog"
+	taste_mult = 1.3
+
+/datum/reagent/toxin/leaper_venom/on_mob_life(mob/living/M)
+	if(volume >= 10)
+		M.adjustToxLoss(5, 0)
+	..()
+
+/mob/living/simple_animal/hostile/jungle/leaper/Initialize()
+	. = ..()
+	verbs -= /mob/living/verb/pulled
+
+/mob/living/simple_animal/hostile/jungle/leaper/CtrlClickOn(atom/A)
+	face_atom(A)
+	target = A
+	if(!isturf(loc))
+		return
+	if(next_move > world.time)
+		return
+	if(hopping)
+		return
+	if(isliving(A))
+		var/mob/living/L = A
+		if(L.incapacitated())
+			BellyFlop()
+			return
+	if(hop_cooldown <= world.time)
+		Hop(player_hop = TRUE)
+
+/mob/living/simple_animal/hostile/jungle/leaper/AttackingTarget()
+	if(isliving(target))
+		return
+	..()
+
+/mob/living/simple_animal/hostile/jungle/leaper/handle_automated_action()
+	if(hopping)
+		return
+	. = ..()
+	if(target)
+		if(isliving(target))
+			var/mob/living/L = target
+			if(L.incapacitated())
+				BellyFlop()
+				return
+		if(!hopping)
+			Hop()
+
+/mob/living/simple_animal/hostile/jungle/leaper/Life()
+	if(ranged_cooldown <= world.time && icon_state != "leaper_alert" && !stat && AIStatus != AI_ON)
+		icon_state = "leaper_alert"
```

Comment:
I think this needs to stay in Life(). handle_automated_action() isn't called if the mob has a CKEY in it, and this icon update is explicitly for player controlled leapers.